app/events/listener.go

```diff
@@ -189,6 +209,56 @@ func (l *TelegramListener) Do(ctx context.Context) error {
 	}
 }
 
+// procNewChatMemberMessage saves new chat member message to locator. It is used to delete the message if the user kicked out
+func (l *TelegramListener) procNewChatMemberMessage(update tbapi.Update) error {
+	fromChat := update.Message.Chat.ID
+	// ignore messages from other chats except the one we are monitor and ones from the test list
+	if !l.isChatAllowed(fromChat) {
+		return nil
+	}
+
+	// ignore multiple new chat members, because we can't delete this message if one of them kicked out
+	if len(update.Message.NewChatMembers) > 1 {
+		log.Printf("[DEBUG] multiple new chat members, ignored")
+		return nil
+	}
+
+	errs := new(multierror.Error)
+
+	for _, member := range update.Message.NewChatMembers {
```

Comment:
this part a little bit confusing. before this the code ignores multiple new chat members, but here it is ranging through all of them

Reply:
It will be correct to get one record. But it's a telegram. therefore, if the list is empty, the loop will not give an error. it can be replaced by checking the equality of the unit and the treatment of the zero index

Reply:
makes sense. I would just add a comment saying something like "we are expecting to get a single message, but ranging over just in case we end up with none".

Reply:
done